archunit/src/main/java/com/tngtech/archunit/lang/ArchRule.java

```diff
@@ -215,6 +222,16 @@ public EvaluationResult evaluate(JavaClasses classes) {
                 return new EvaluationResult(this, events, priority);
             }
 
+            private void verifyInputNotEmpty(Iterable<T> allObjects) {
+                if (isVerificationForEmptyInputActive() && Iterables.isEmpty(allObjects)) {
+                    throw new AssertionError("The input for a rule should not be empty, since then the test will always be green and does not really test something");
```

Comment:
Renamed it.

What do you mean with _hint_, adding some documentation to the user guide? I think that's a good idea, but I have a few questions. Where exactly should I add it? Probably adding a new section in the "Advanced Configuration" section?

By looking at the code, for me, it seems, that the user guide is kind of duplicated. There exists the `index.html` which includes everything from the user guide, but then there are also the `.adoc` files which contain the same information. What exactly is there the difference?

Reply:
By "hint" I meant add it to the exception message :wink: I think it's nice when the exception message already tells you how you can change the behavior in this case.
Nevertheless, it might also be a good idea to add it to the user guide. The HTML is generated and should never be modified manually, the source of truth is the `adoc` (the reason is that GitHub pages doesn't whitelist the Jekyll Asciidoc plugin :disappointed: Otherwise, the HTML could be generated on the fly on page deployment).
So yes, the "Advanced Configuration" section would also be my intuition where I would put it, feel free to add it to the `adoc` file! The HTML you can ignore for now, it will be updated during the release process anyway.